src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -289,37 +289,59 @@ internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList
             var newBasesBeingResolved = basesBeingResolved.Prepend(this.OriginalDefinition);
             var baseInterfaces = ArrayBuilder<NamedTypeSymbol>.GetInstance();
 
-            NamedTypeSymbol baseType = null;
+            TypeWithAnnotations baseType = default;
```

Comment:
>TypeWithAnnotations [](http://example.com/codeflow?start=12&length=19)

Is the type change necessary? Do we allow nullable annotations at the top level for a base type?

Reply:
It is not necessary. After a little thought I think the right thing is probably to continue to discard any nullable annotation found from binding the base type (we give an error diagnostic if the `?` annotation is used in the base type) and effectively only check for nested nullability differences.